### PR TITLE
[Modular] Energy fire axes and DS-2 atmos QoLs

### PR DESCRIPTION
**Pull request by Ebin-Halcyon**
## About The Pull Request

Adds in an energy Fire axe for use on DS-2/Interdyne since Bone axes were cut off from them completely, rendering them unable to get axes.

No, these are not in the uplink and I don't plan to move them to the uplink, if you want to do it yourself, be my guest.

Also adds in a few porta scrubbers/porta pumps for use in atmos fuckery/fuckups since DS-2 didn't have them.

images of said axe.
Wielded
Mmr6bn2
Unwielded
image

In hands
hu0JieC
K6pxnGi

## How This Contributes To The Skyrat Roleplay Experience

Gives the Syndicate roles an alternative to bone axes (that they can no longer acquire) along with more tools for Atmos projects.

## Changelog

:cl:
add: A shipment of energy fire axes, portascrubbers, and porta pumps have been sent to DS-2
/:cl:

### CODE REVIEW

**Comment by Ebin-Halcyon** [comment]
Guh, fixing CI

**Comment by softcerv** [comment]
This looks sick!

**Comment by Ebin-Halcyon** [comment]
There we go, no more runtimes.

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	w_class = WEIGHT_CLASS_BULKY
+	light_system = MOVABLE_LIGHT
+	light_range = 6
+	light_color = COLOR_SOFT_RED
+	light_on = FALSE
+	attack_verb_continuous = list("attacks", "chops", "cleaves", "tears", "lacerates", "cuts")
+	attack_verb_simple = list("attack", "chop", "cleave", "tear", "lacerate", "cut")
+	armour_penetration = 25
+	sharpness = SHARP_EDGED
+	special_desc_requirement = EXAMINE_CHECK_SYNDICATE_TOY
```
Comment (YakumoChen) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: Make this a normal check, not the TOY check please.

**Comment by YakumoChen** [comment]
I strongly recommend merging this into whatever main DS2 module we have instead of making another module just for the axe. At some point we're gonna need to collapse a bunch of modular content into alike folders so its easier in general to browse.

**Comment by Ebin-Halcyon** [comment]
> I strongly recommend merging this into whatever main DS2 module we have instead of making another module just for the axe. At some point we're gonna need to collapse a bunch of modular content into alike folders so its easier in general to browse.

There isn't a main module for DS-2 and i'm not a good enough coder to compile everything into one module without breaking everything else.

**Comment by YakumoChen** [comment]
Really just a matter of making sure the file names on tgstation.dm match, you could put the contained code anywhere in the codebase and it would work.

But if that's the case someonene else will have to do it.

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+/obj/item/fireaxe/energy
+	icon = 'modular_skyrat/master_files/icons/obj/energy_axe.dmi'
+	icon_state = "energy_axe0"
+	base_icon_state = "energy_axe"
+	lefthand_file = 'modular_skyrat/master_files/icons/mob/energyaxe_lefthand.dmi'
+	righthand_file = 'modular_skyrat/master_files/icons/mob/energyaxe_righthand.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/back.dmi'
+	name = "energy fire axe"
+	desc = "You aren't quite sure if this counts as a fire axe anymore, but it sure is fancy! A tag hangs off of it reading; 'properly of the Gorlex Marauders'"
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]:
```suggestion
	desc = "You aren't quite sure if this counts as a fire axe anymore, but it sure is fancy! A tag hangs off of it reading: 'properly of the Gorlex Marauders'"
```

improper use of a semi-colon

**Comment by Ebin-Halcyon** [comment]
Can't believe ratfish broke CI.

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/obj/item/fireaxe/energy
+	icon = 'modular_skyrat/master_files/icons/obj/energy_axe.dmi'
+	icon_state = "energy_axe0"
+	base_icon_state = "energy_axe"
+	lefthand_file = 'modular_skyrat/master_files/icons/mob/energyaxe_lefthand.dmi'
+	righthand_file = 'modular_skyrat/master_files/icons/mob/energyaxe_righthand.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/back.dmi'
+	name = "energy fire axe"
+	desc = "You aren't quite sure if this counts as a fire axe anymore, but it sure is fancy! A tag hangs off of it reading: 'properly of the Gorlex Marauders'"
+	force = 5
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm:12]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	throwforce = 15
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm:13]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	w_class = WEIGHT_CLASS_BULKY
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	light_system = MOVABLE_LIGHT
+	light_range = 6
+	light_color = COLOR_SOFT_RED
+	light_on = FALSE
+	attack_verb_continuous = list("attacks", "chops", "cleaves", "tears", "lacerates", "cuts")
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+	attack_verb_simple = list("attack", "chop", "cleave", "tear", "lacerate", "cut")
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	armour_penetration = 35
+	sharpness = SHARP_EDGED
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+	special_desc_requirement = EXAMINE_CHECK_SYNDICATE
+	special_desc = "An energized fireaxe of Syndicate descent."
+	max_integrity = 200
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 100, ACID = 30)
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+	resistance_flags = FIRE_PROOF
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+	wound_bonus = -15
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+	bare_wound_bonus = 20
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: unneeded

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+/obj/item/fireaxe/energy/proc/energy_wield(obj/item/source, mob/living/carbon/user)
+	SIGNAL_HANDLER
+
+	wielded = TRUE
+	hitsound = 'sound/weapons/blade1.ogg'
+	START_PROCESSING(SSobj, src)
+	set_light_on(TRUE)
+
+//Swap hitsounds into the smack sound
```
Comment (Zonespace27) [inline modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: ?

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm (hunk 14 above)
Comment (Ebin-Halcyon) [reply modular_skyrat/modules/energy_axe/code/energy_fireaxe.dm]: Sets the lights of the energy axe on or off, swaps it from the energy sword attack sound to the basic body hit sound.

**Pull request merged**